src/ooni/utils.hpp

```diff
@@ -20,9 +20,39 @@ void ip_lookup(Callback<Error, std::string> callback, Settings settings = {},
                Var<Reactor> reactor = Reactor::global(),
                Var<Logger> logger = Logger::global());
 
+void resolver_lookup(Callback<Error, std::string> callback, Settings settings = {},
+                     Var<Reactor> reactor = Reactor::global(),
+                     Var<Logger> logger = Logger::global());
+
+class IPLocation {
+    public:
+      IPLocation(std::string path_country, std::string path_asn);
+
+      ~IPLocation();
+
+      ErrorOr<std::string> resolve_country_code(std::string ip);
+
+      ErrorOr<std::string> resolve_country_name(std::string ip);
+
+      ErrorOr<std::string> resolve_asn(std::string ip);
+
+    private:
+      GeoIP *gi_asn = nullptr;
+      GeoIP *gi_country = nullptr;
+      Var<Logger> logger = Logger::make();
```

Comment:
I think it would be interesting to log failures in the context of a specific test; as such, I think the `logger` should not be a class member, rather it should be passed to each of the `resolve_xx` methods as the last optional argument otherwise initialized to the default logger.